src/components/Route.js

```diff
@@ -0,0 +1,14 @@
+import { extractParams, extractQuery } from '../utils'
+
+export const Route = ({ exact, path = '', view }) => {
+  const match = exact
```

Comment:
@lukejacksonn The Route component tries to match the pathname with the given path (and if there is match extractParams and extractQuery) **every time**.

How do you feel about that? Seems wasteful. I think a `Router` parent component could help ameliorate this. What is the advantage of only `Route`?

Reply:
Yes, match gets calculated every render, not sure how expensive it is but I assume the overhead is less than or the same as that of calling `h` which doesn't really concern me. The more expensive function `extractParams` and `extractQuery` only get called when required (match is true). If perf was a concern here you could memoize any of the 3 functions, but I don't see a lot of gain to be had.

The benefit you get from dynamic routing like this, rather than static routing like this old router is flexibility and composability. Routing is a hard problem and there are often complex requirements; not always a simple 1-to-1 mappings of routes to views. If you can meet these requirements with various configurations of very simple components like `Route`, `Switch` and `Link` then why introduce another thing?

A parent `Router` component implies you are to be confined within that component, it basically takes over your app, becoming the top component. It demands nesting of children but could it be nested itself like a simple route? I have not thought about it too much because I haven't come across a need for one. How does it help?